### PR TITLE
USDX Bridge

### PR DESCRIPTION
**Pull request by 0xCooki**
- USDX Bridge contract
- USDX Bridge tests
- Added mock custom gas token and mock stablecoins (USDC, USDT, DAI) to testing environment
- Added USDX Bridge to deployment scripts

### CODE REVIEW

File: packages/contracts-bedrock/src/L1/USDXBridge.sol [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    uint256 public depositCap;
+
+    /// @notice The total amount of USDX bridged via this contract.
+    uint256 public totalBridged;
+
+    /// INITIALIZE ///
+
+    constructor() {
+        initialize({
+            _owner: address(0xdEaD),
```
Comment (jacko125) [inline packages/contracts-bedrock/src/L1/USDXBridge.sol:49]: Do we want the owner to be set / initialised to deployer?

File: packages/contracts-bedrock/src/L1/USDXBridge.sol [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    /// @param  _to Recieving address on L2.
+    function bridge(
+        address _stablecoin,
+        uint256 _amount,
+        address _to
+    ) external nonReentrant {
+        /// Checks
+        require(allowlisted[_stablecoin], "USDXBridge: Stablecoin not accepted.");
+        require(_amount > 0, "USDXBridge: May not bridge nothing.");
+        uint256 bridgeAmount = _getBridgeAmount(_stablecoin, _amount);
```
Comment (jacko125) [inline packages/contracts-bedrock/src/L1/USDXBridge.sol:101]: Good check here for conversion between 6 decimal places for USDT / USDC and USDX should be same.

**Comment by jacko125** [comment]
Looks good USDX bridge contract and comments are good. Just one question on the constructor / initialize() function

File: packages/contracts-bedrock/src/L1/USDXBridge.sol [hunk 3]
```diff
@@ -36,6 +36,12 @@ contract USDXBridge is OwnableUpgradeable, ReentrancyGuard, ISemver {
     /// @notice The total amount of USDX bridged via this contract.
     uint256 public totalBridged;
 
+    /// @notice An event emitted when a bridge deposit is made by a user.
+    event BridgeDeposit(address indexed _stablecoin, uint256 indexed _amount, address indexed _to);
```
Comment (jacko125) [inline packages/contracts-bedrock/src/L1/USDXBridge.sol]: Does it make sense to index the `_amount` field?

File: packages/contracts-bedrock/src/L1/USDXBridge.sol [hunk 4]
```diff
@@ -36,6 +36,12 @@ contract USDXBridge is OwnableUpgradeable, ReentrancyGuard, ISemver {
...
+
+    /// @notice An event emitted when an ERC20 token is withdrawn from this contract.
+    event WithdrawCoins(address indexed _coin, uint256 indexed _amount, address indexed _to);
```
Comment (jacko125) [inline packages/contracts-bedrock/src/L1/USDXBridge.sol]: Same for WithdrawCoins event

**Comment by jacko125** [comment]
@0xCooki One comment on the events emitting

File: packages/contracts-bedrock/src/L1/USDXBridge.sol (hunk 3 above)
Comment (0xCooki) [reply packages/contracts-bedrock/src/L1/USDXBridge.sol]: Will remove the `indexed` modifier

**Review by jacko125 — APPROVED** [review]

**Pull request merged**